pkg/plugin/api/server.go

```diff
@@ -337,3 +346,16 @@ func (c *grpcServer) SendAlert(ctx context.Context, in *proto.AlertRequest) (*pr
 	c.service.SendAlert(ctx, in.ClientID, alert)
 	return &proto.Empty{}, nil
 }
+
+func extractObjectStoreMetadata(ctx context.Context) context.Context {
+	// Second value is ignored as metadata is always set by grpc.
+	md, _ := metadata.FromIncomingContext(ctx)
+	for k, v := range md {
+		if strings.HasPrefix(k, "x-octant-") {
```

Comment:
This is something up for discussion. grpc metadata does not allow strings to be wrapped by type and strictly accept string. This is the convention i used here to avoid octant from loading header like `content-type`, `user-agent` into the context which are subject to change in the future.